test/comments.d

```diff
@@ -0,0 +1,82 @@
+import app;
+import utils;
+
+import vibe.d;
+
+// existing dlang bot comment -> update comment
+unittest
+{
+    setAPIExpectations(
+        "/github/repos/dlang/phobos/pulls/4921/commits",
+        "/github/repos/dlang/phobos/issues/4921/comments",
+        "/bugzilla/buglist.cgi?bug_id=8573&ctype=csv&columnlist=short_desc",
+        "/github/repos/dlang/phobos/issues/comments/262784442",
+        (scope HTTPServerRequest req, scope HTTPServerResponse res){
```

Comment:
That's fine and also comes with clearer error messages.

Technically it's possible to pass a polymorphic lambda to a template and check whether it's instantiatable though.
The error messages are not so good, b/c you can't know which type was meant.

```d
void foo(Args...)()
{
    static if (__traits(compiles, Args[0](12)))
    {
        pragma(msg, "int");
    }
    else static if (__traits(compiles, Args[0]("string")))
    {
        pragma(msg, "string");
    }
}

void test()
{
    foo!(a => 2 * a);
    foo!(a => a ~ "foo");
}
```d